gluten-substrait/src/main/scala/org/apache/spark/sql/execution/GlutenFallbackReporter.scala

```diff
@@ -40,7 +40,7 @@ case class GlutenFallbackReporter(glutenConf: GlutenConfig, spark: SparkSession)
       return plan
     }
     printFallbackReason(plan)
-    if (glutenConf.glutenUiEnabled) {
+    if (GlutenUIUtils.uiEnabled(spark.sparkContext)) {
```

Comment:
Can we use another method to check if the UI is enabled? We need to use `glutenConf.glutenUiEnabled` to prevent user configuration at runtime.

Reply:
the gluten ui config cannot be modified at runtime. I'm not quite sure what you mean. Please correct me if I'm miss something.

Reply:
the most accurate approach would be to check if the tabs in the webUI include GlutenSQLTab instance, but perhaps it's a bit heavy? I'm fine with both.

Reply:
`spark.sparkContext.ui.isDefined && glutenConf.glutenUiEnabled` should be enough

Reply:
this is not enough. `uiEnabled` might be called anywhere, and when it is called, `sqlConf` might not have been initialized yet, such as during plugin initialization.